lnst/Recipes/ENRT/ConfigMixins/DisableIdleStatesMixin.py

```diff
@@ -31,7 +32,7 @@ def apply_sub_configuration(self, config):
         super().apply_sub_configuration(config)
 
         latency = getattr(self.params, "minimal_idlestates_latency", None)
-        if latency is not None:
+        if latency is not None and latency >= 0:
```

Comment:
i'm not sure i understand why this is necessary?

you say that
> recipes with `DisableIdleStatesMixin` deep in the inheritance tree.

but `self.params` are exposed to the top level of the Recipe... so you can always override it with a `=None` value? and using a `<0` value is the same - you basically have to at the top level do `minimal_idlestates_latency=-1`... but you can just as well do `minimal_idlestates_latency=None` right?

Reply:
parameters are optional by default and this one doesn't say `mandatory=True` so i guess this is already the case...

so if you don't specify the parameter you get `None` as the default returned from `getattr(self.params, "minimal_idlestates_latency", None)`

so then the question now becomes... why provide `minimal_idlestates_latency=-1` when you can simply skip the parameter?

Reply:
Yes, you can simply omit the parameter for the recipe and it would be skipped. See the example of a test that does exactly that, internal uuid is 10050e83-714c-4323-8d22-1199cde85e55.

The recipe prints: `configuration of idle states skipped`